exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -0,0 +1,245 @@
+import static java.util.Collections.singletonList;
+
+import com.exonum.binding.blockchain.Block;
+import com.exonum.binding.proxy.NativeHandle;
+import com.exonum.binding.runtime.ReflectiveModuleSupplier;
+import com.exonum.binding.service.BlockCommittedEvent;
+import com.exonum.binding.service.Service;
+import com.exonum.binding.service.ServiceModule;
+import com.exonum.binding.service.adapters.UserServiceAdapter;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+
+/**
+ * TestKit for testing blockchain services. It offers simple network configuration emulation
+ * (with no real network setup). Although it is possible to add several validator nodes to this
+ * network, only one node will create the service instances and will execute their operations
+ * (e.g., {@link Service#afterCommit(BlockCommittedEvent)} method logic).
+ *
+ * <p>When TestKit is created, Exonum blockchain instance with given services is initialized and
+ * genesis block is committed. The service instances are created and their
+ * {@linkplain UserServiceAdapter#initialize(long)}  initialization} methods are called and
+ * {@linkplain UserServiceAdapter#mountPublicApiHandler(long)} public API handlers} are created.
+ *
+ * @see <a href="https://exonum.com/doc/version/0.10/get-started/test-service/">TestKit documentation</a>
+ */
+public final class TestKit {
+
+  @VisibleForTesting
+  static final short MAX_SERVICE_NUMBER = 256;
+  private final Injector frameworkInjector = Guice.createInjector(new TestKitFrameworkModule());
+
+  private final NativeHandle nativeHandle;
+  private final Map<Short, Service> services = new HashMap<>();
+
+  private TestKit(List<Class<? extends ServiceModule>> serviceModules, EmulatedNodeType nodeType,
+                  short validatorCount, @Nullable TimeProvider timeProvider) {
+    List<UserServiceAdapter> serviceAdapters = toUserServiceAdapters(serviceModules);
+    populateServiceMap(serviceAdapters);
+    boolean isAuditorNode = nodeType == EmulatedNodeType.AUDITOR;
+    UserServiceAdapter[] userServiceAdapters = serviceAdapters.toArray(new UserServiceAdapter[0]);
+    // TODO: fix after native implementation
+    nativeHandle = null;
+//    nativeHandle = new NativeHandle(
+//        nativeCreateTestKit(userServiceAdapters, isAuditorNode,
+//            validatorCount, timeProvider));
+  }
+
+  /**
+   * Creates a TestKit network with a single validator node for a single service.
+   */
+  public static TestKit forService(Class<? extends ServiceModule> serviceModule) {
+    return new TestKit(singletonList(serviceModule), EmulatedNodeType.VALIDATOR, (short) 0, null);
+  }
+
+  /**
+   * Returns an instance of a service with the given service id and service class.
```

Comment:
\+ "Only user-defined services can be requested, i.e., it is not possible to get an instance of a built-in service such as the time oracle."